Upstream-Builder Container Image Build Failing

Hi Team,
I am trying to build the upstream-builder container image, but the build is failing with `fatal error: linux/limits.h: No such file or directory`. 

```
unshare.c:21:10: fatal error: linux/limits.h: No such file or directory
   21 | #include <linux/limits.h>
      |          ^~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:57: bin/opm] Error 1
Error: building at STEP "RUN make static": while running runtime: exit status 2
```

To fix this, need to install `linux-headers` in [upstream-builder.Dockerfile](https://github.com/operator-framework/operator-registry/blob/master/upstream-builder.Dockerfile)

--- a/upstream-builder.Dockerfile
+++ b/upstream-builder.Dockerfile
@@ -1,6 +1,6 @@
 FROM golang:1.23-alpine as builder
 
-RUN apk update && apk add sqlite build-base git mercurial bash
+RUN apk update && apk add sqlite build-base git mercurial bash linux-headers
 WORKDIR /build
 
 COPY . .

